tinygrad/device.py

```diff
@@ -241,13 +241,13 @@ def _interpret_ast(ast:LazyOp) -> str:
 # **************** for Compiled Devices ****************
 
 class CompiledASTRunner(JITRunner):
-  def __init__(self, ast:Optional[LazyOp], name:str, prg:str, lib:bytes, global_size:Optional[List[int]]=None, local_size:Optional[List[int]]=None):  # noqa: E501
+  def __init__(self, ast:Optional[LazyOp], name:str, prg:str, lib:bytes, device:Compiled, global_size:Optional[List[int]]=None, local_size:Optional[List[int]]=None):  # noqa: E501
```

Comment:
Why is this new parameter required? If it really is, can you add it in a separate PR?

Reply:
I need a way to get from a JITRunner to the device that should be used to graph it, I couldn't find a way without adding this. Could go to a different PR but only the jit Grapher needs this atm

Reply:
I have a fix for this in master, the device is in clprg.